### ISSUE TITLE
Participation in the node-chakracore Working Group

### ISSUE THREAD

**Issue by orangemocha** [nodejs/node-chakracore#18]
Following up to the discussion at https://github.com/nodejs/node/pull/4765, I’d like to start a new working group, [node-chakracore](https://github.com/orgs/nodejs/teams/node-chakracore), and extend an invitation to members of @nodejs/collaborators, @nodejs/ctc and @nodejs/chakra to participate. 

As I see it, the main purpose of this WG would be to act in the same capacity for this repo as @nodejs/collaborators act for the nodej/node repo, primarily reviewing issues and pull requests (with potential escalation to the CTC). 

Members of the Microsoft Chakra team who already worked on node-chakracore will naturally be part of the WG. Input from Node collaborators would be very valuable to ensure that the work on this project can continue in a way that it’s in line with the way that things are done in nodejs/node, and to socialize the Chakra team with Node collaborators. 

With the assumption that changes to deps/chakrashim can be treated as atomic dependency updates, the rate of changes to node source code is expected to be pretty low, so hopefully this won’t be a big time drain. I don’t expect that regular meetings will be needed either.

Please respond here if you would like to be a member of the @nodejs/node-chakracore team. Thanks!

**Comment by cjihrig** [comment]
@orangemocha is there a timeline for support of non-Windows operating systems?

**Comment by kunalspathak** [comment]
@cjihrig , We are working actively to port Chakracore on linux and you can follow its progress [here](https://github.com/Microsoft/chakracore/tree/linux). We should have bare minimum working after couple of months. At that time we will work to get the necessary work to make node.js + chakracore work on linux.

**Comment by cjihrig** [comment]
Thanks @kunalspathak

**Comment by joshgav** [comment]
Count me in!

**Comment by thefourtheye** [comment]
Well, I would love to review the code patches submitted. Please count me in.

I assume committing directly without a PR is not encouraged in this repo as well. Will the PRs submitted follow the similar procedure we follow in the nodejs core?

**Comment by orangemocha** [comment]
I have given this a bit more thought and since the main goal here is to get as much input as possible from Node.js collaborators, we don’t need to create an additional barrier for them. So I think it would be appropriate to give all @nodejs/collaborators and @nodejs/ctc members collaborator status to this repo (i.e. write permission + ability to sign off pull requests). 

One possible counterargument is that people might not feel qualified to sign off PRs here if they are not following the development of node-chakracore closely. But Node.js is already a vast project and not all collaborators are equally experienced in all areas, so this is nothing new and the [collaborator’s guide](https://github.com/nodejs/node/blob/master/COLLABORATOR_GUIDE.md#accepting-modifications) already states:

> All pull requests must be reviewed and accepted by a Collaborator _with sufficient expertise who is able to take full responsibility for the change._ 

So basically we would be saying: come here if you please and review what you feel comfortable with. Does anyone see any problems with this approach?

**Comment by orangemocha** [comment]
@thefourtheye, thank you! In any case I have added you to @ nodejs/node-chakracore. Yes, PRs should  follow the similar procedure we follow in the nodejs core. That includes running CI on all PRs. We are also working on adding a node-chakracore specific CI job that will actually test the chakracore build.

@joshgav, thank you for the request! As already discussed offline, while I would certainly vouch for you personally, we should follow the same rules here as we do for nodejs/node. Per [Governance Policy](https://github.com/nodejs/node/blob/master/GOVERNANCE.md#collaborators):

> Individuals making significant and valuable contributions are made Collaborators and given commit-access to the project. 

(.. which in the context of this repo I think should also include contributions to node-chakracore.) 

Please do contribute by means of issues, pull requests, and reviews and once the requirement is met we will certainly give you the commit bit.

**Comment by thefourtheye** [comment]
@orangemocha Thank you :-)

**Comment by joshgav** [comment]
Makes sense to me, thanks!

**Comment by jbergstroem** [comment]
@orangemocha good point. As a member of the build group I'd also like to know more about changes that might affect how we build node. Unfortunately, most changes in the build system takes a long time to sunset or integrate (ci, whatnot).

**Comment by rvagg** [comment]
@orangemocha makes perfect sense, we already have a culture of _don't touch that thing if you don't understand it_ so I don't imagine any problems here.

**Comment by joaocgreis** [comment]
If I understood it correctly, the @nodejs/chakra team is composed by people who understand the Chakra engine in detail, and @nodejs/node-chakracore by people who care about the integration of Chakra with node. So we should mention @nodejs/chakra for complex engine issues and @nodejs/node-chakracore for everything else. Is this correct?

**Comment by orangemocha** [comment]
@joaocgreis that was the idea. If this proves to be confusing we could rename @ nodejs/chakra to something like @ nodejs/chakra-internals.

**Comment by rvagg** [comment]
`@nodejs/chakra` has an obvious parallel with `@nodejs/v8` which is used for exactly the purpose being described here and `@nodejs/node-chakracore` has a temporary look to it so this all makes sense I think.

**Comment by thefourtheye** [comment]
Wouldn't it be better if we reverse the names? Chakra core will have core people and chakra will have other contributors. How is it?

**Comment by kunalspathak** [comment]
>  Chakra core will have core people and chakra will have other contributors. 

This will still be confusing because `@nodejs/chakra` has description of `chakra` as well as `chakracore`. I feel that current nomenclature is fine. 
- `@nodejs/chakra` : People who understand engine internals up to (but not limited to) chakra shim layer.
- `@nodejs/node-chakracore` : People who understand from chakra shim layer up to (and again not limited to) the integration in node.

**Comment by orangemocha** [comment]
> Wouldn't it be better if we reverse the names? Chakra core will have core people and chakra will have other contributors. How is it?

I am guessing a source of confusion here is the `core` suffix. It doesn't refer to a core team. The engine that has been added in this fork is called 'chakracore'. 'Chakra' is the name of the closed source distribution which includes chakracore with a few additional features, and it has also been historically the name of the team at Microsoft who works on the engine. 

To avoid this confusion, I think it might be a good idea to rename `@nodejs/chakra` to `@nodejs/chakracore`, and consistently refer to 'chakracore' instead of 'chakra' in this repo.

**Comment by thefourtheye** [comment]
> To avoid this confusion, I think it might be a good idea to rename @nodejs/chakra to @nodejs/chakracore, and consistently refer to 'chakracore' instead of 'chakra' in this repo.

+1

**Comment by orangemocha** [comment]
Thanks everyone for the input. I made the following changes:
1. Renamed the Chakra team to ChakraCore. So we have:
   - [@nodejs/node-chakracore](https://github.com/orgs/nodejs/teams/node-chakracore) : people involved in the integration of node with chakracore, a.k.a. maintainers of this repo.
   - [@nodejs/ChakraCore](https://github.com/orgs/nodejs/teams/chakracore) : ChakraCore engine internals  experts.
2. Given @nodejs/collaborators write access to this repo.
3. Opened [a PR](https://github.com/nodejs/node-chakracore/pull/27) to clarify collaborator and contribution guidelines, including treating @nodejs/collaborators as project collaborators in this repo so that they can sign off pull requests.
